### PR TITLE
QGC Parameters

### PR DESCRIPTION
**Pull request by afwilkin**
When you want to download a parameter file... it can not be directly uploaded into QGC. This givces you the option to download the QGC compatible one. This PR was based off of @farhangnaderi PR here: #264

### CODE REVIEW

**Comment by afwilkin** [comment]
@bkueng let me know if there is any changes you would like me to make

**Comment by farhangnaderi** [comment]
Thanks Andrew! I appreciate it!

File: app/plot_app/templates/header.html [hunk 1]
```diff
@@ -56,7 +56,9 @@ <h1>
 			<div class="dropdown-menu">
 				<a class="dropdown-item" href="download?log={{ log_id }}">Log File</a>
 				<a class="dropdown-item" href="download?log={{ log_id }}&type=1" target="_blank">Parameters</a>
```
Comment (bkueng) [inline app/plot_app/templates/header.html:58]: I think we can remove this option (but keep the implementation), and just offer the QGC versions.

File: app/tornado_handlers/download.py [hunk 2]
```diff
@@ -155,6 +154,75 @@ def kml_colors(flight_mode):
...
+        elif download_type == '4': # download QGC parameters
+            ulog = load_ulog_file(log_file_name)
+            param_keys = sorted(ulog.initial_parameters.keys())
+
+            self.set_header("Content-Type", "text/plain")
+            self.set_header('Content-Disposition', 'inline; filename=vehicle.params')
+
+            delimiter = '	'
+            for param_key in param_keys:
+                self.write("1")
```
Comment (bkueng) [inline app/tornado_handlers/download.py:166]: Can you add a comment that this is the sysid (and compid below)?

File: app/tornado_handlers/download.py [hunk 3]
```diff
@@ -155,6 +154,75 @@ def kml_colors(flight_mode):
...
+                                self.write("6")
+                            else:
+                                self.write(delimiter)
+                                self.write("9")
+
+                            self.write('\n')
+                    except:
+                        pass
+
             else:
```
Comment (bkueng) [inline app/tornado_handlers/download.py:226]: This `else` case was part of the `elif download_type == '3': # download the non-default parameters` case above and needs to be copied up to that case again, and here it needs to be adjusted for the QGC format.

**Review by bkueng — COMMENTED** [review]
There's also a couple of CI problems. We should start to split the long method into individual methods by download type.

**Comment by afwilkin** [comment]
Apologies Beat, I did not see you had reviewed this. I will fix these

**Comment by afwilkin** [comment]
@bkueng I addressed your comments. I know you mentioned there are some CI problems that we will incur. Mainly the long method. what is the method length that will pass the CICD test so I know how to modify the code?

**Comment by afwilkin** [comment]
@bkueng made changes. 

Also added something to the bottom of the readme to let others know to run `./run_pylint` before making PR to help with CICD.

**Review by bkueng — APPROVED** [review]
Thanks

**Pull request merged**